src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -753,4 +830,152 @@ public Transaction update(final Transaction transaction) {
 		this.getSession().flush();
 		return transaction;
 	}
+
+	public long countAllStudyTransactions(final Integer studyId, final StudyTransactionsRequest studyTransactionsRequest) {
```

Comment:
param seems not to be used. Do we need to different functions? we usually assume that if the object to filter is null, then it returns or count everything without filtering.